python/ray/data/_internal/arrow_block.py

```diff
@@ -414,37 +415,34 @@ def sum_of_squared_diffs_from_mean(
     def sort_and_partition(
         self, boundaries: List[T], key: "SortKeyT", descending: bool
     ) -> List["Block"]:
-        if len(key) > 1:
-            raise NotImplementedError(
-                "sorting by multiple columns is not supported yet"
-            )
-
         if self._table.num_rows == 0:
             # If the pyarrow table is empty we may not have schema
             # so calling sort_indices() will raise an error.
             return [self._empty_table() for _ in range(len(boundaries) + 1)]
 
         context = DataContext.get_current()
         sort = get_sort_transform(context)
-        col, _ = key[0]
+        columns = [k[0] for k in key]
         table = sort(self._table, key, descending)
         if len(boundaries) == 0:
             return [table]
 
-        partitions = []
         # For each boundary value, count the number of items that are less
         # than it. Since the block is sorted, these counts partition the items
         # such that boundaries[i] <= x < boundaries[i + 1] for each x in
         # partition[i]. If `descending` is true, `boundaries` would also be
         # in descending order and we only need to count the number of items
         # *greater than* the boundary value instead.
+        table_items = [tuple(d.values()) for d in _to_pylist(table.select(columns))]
```

Comment:
This looks heavy. I guess it's possible to bisect on the original arrow table, without converting it to a pylist.

Reply:
Would obviously prefer to bisect on the entire arrow table but not sure how you are suggesting to do it here?

Reply:
I was thinking of doing the bisect with the original arrow table, with a custom key argument.